### PR TITLE
Source Marketo: set high test strictness level

### LINKED ISSUE #19055: Source Marketo: enable `high` test strictness level in SAT
## What
A `test_strictness_level` field was introduced to Source Acceptance Tests (SAT).
Marketo is a generally_available connector, we want it to have a `high` test strictness level.

**This will help**:
- maximize the SAT coverage on this connector.
- document its potential weaknesses in term of test coverage.

## How
1. Migrate the existing `acceptance-test-config.yml` file to the latest configuration format. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/source-acceptance-test/README.md#L61))
2. Enable `high` test strictness level in `acceptance-test-config.yml`. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/docs/connector-development/testing-connectors/source-acceptance-tests-reference.md#L240))
3. Commit changes on `acceptance-test-config.yml` and open a PR.
4. Run SAT with the `/test` command on the branch.
5. If tests are failing please fix the failing test or use `bypass_reason` fields to explain why a specific test can't be run.

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/airbyte/issues/19055

## How
- Modify full_refresh catalog
- Modify incremental catalog
- Add expected records
- Refactor acceptance test config

### CODE REVIEW

File: airbyte-integrations/connectors/source-marketo/acceptance-test-config.yml [hunk 1]
```diff
@@ -1,29 +1,81 @@
...
-      configured_catalog_path: "integration_tests/configured_catalog.json"
-      empty_streams: ["lists", "campaigns"]
-      timeout_seconds: 3600
-      expect_records:
-        path: "integration_tests/expected_records.jsonl"
+    tests:
+      - config_path: "secrets/config.json"
+        expect_records:
+          path: "integration_tests/expected_records.jsonl"
+        empty_streams:
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-marketo/acceptance-test-config.yml:23]: @igrankova managed to populate 29 streams, but another 23 streams can not be populated without leads/customers/etc interaction

**Review by darynaishchenko — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-marketo/acceptance-test-config.yml [hunk 2]
```diff
@@ -1,29 +1,81 @@
...
+          - name: "activities_change_revenue_stage_manually"
+            bypass_reason: "Marketo does not provide a way to populate this stream without outside interaction"
+          - name: "activities_receive_sales_email"
+            bypass_reason: "Marketo does not provide a way to populate this stream without outside interaction"
+          - name: "activities_replyto_sales_email"
+            bypass_reason: "Marketo does not provide a way to populate this stream without outside interaction"
+          - name: "activities_addto_opportunity"
+            bypass_reason: "Marketo does not provide a way to populate this stream without outside interaction"
+          - name: "activities_open_email"
+            bypass_reason: "Marketo does not provide a way to populate this stream without outside interaction"
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-marketo/acceptance-test-config.yml:69]: dang.

File: airbyte-integrations/connectors/source-marketo/integration_tests/expected_records.jsonl [hunk 3]
```diff
@@ -1,63 +1,153 @@
...
...alSearchPhrase": null, "originalReferrer": null, "emailInvalid": false, "emailInvalidCause": null, "unsubscribed": false, "unsubscribedReason": null, "doNotCall": false, "mktoDoNotCallCause": null, "doNotCallReason": null, "marketingSuspended": false, "marketingSuspendedCause": null, "blackListed": false, "blackListedCause": null, "mktoPersonNotes": null, "anonymousIP": null, "inferredCompany": null, "inferredCountry": null, "inferredCity": null, "inferredStateRegion": null, "inferredPostalCode": null, "inferredMetropolitanArea": null, "inferredPhoneAreaCode": null, "emailSuspended": null, "emailSuspendedCause": null, "emailSuspendedAt": null, "department": null, "createdAt": "2023-01-23T08:20:01Z", "updatedAt": "2023-01-23T08:20:01Z", "cookies": null, "externalSalesPersonId": null, "leadPerson": "898", "leadRole": null, "leadSource": null, "leadStatus": null, "leadScore": null, "urgency": null, "priority": null, "relativeScore": null, "relativeUrgency": null, "rating": null, "personPrimaryLeadInterest": "898", "leadPartitionId": "1", "leadRevenueCycleModelId": null, "leadRevenueStageId": null, "acquisitionProgramId": null, "mktoAcquisitionDate": null}, "emitted_at": 1674502167300}
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-marketo/integration_tests/expected_records.jsonl:33]: 👍 fake email addresses

File: airbyte-integrations/connectors/source-marketo/integration_tests/expected_records.jsonl [hunk 4]
```diff
@@ -1,63 +1,153 @@
...
+{"stream": "activities_delete_lead", "data": {"marketoGUID": "2280", "leadId": 849, "activityDate": "2023-01-23T08:36:22Z", "activityTypeId": 37, "campaignId": 1162, "primaryAttributeValueId": "849", "primaryAttributeValue": "Kataldar-70"}, "emitted_at": 1674506602584}
+{"stream": "activities_delete_lead", "data": {"marketoGUID": "2281", "leadId": 852, "activityDate": "2023-01-23T08:36:22Z", "activityTypeId": 37, "campaignId": 1162, "primaryAttributeValueId": "852", "primaryAttributeValue": "Kataldar-74"}, "emitted_at": 1674506602585}
+{"stream": "activities_email_bounced", "data": {"marketoGUID": "2273", "leadId": 880, "activityDate": "2023-01-23T08:25:26Z", "activityTypeId": 8, "campaignId": 1159, "primaryAttributeValueId": "1009", "primaryAttributeValue": "Test Nurture.integration-tgest@airbyte.io", "campaign_run_id": 112, "category": "2", "email": "alastor@moody.com", "details": "550 5.4.1 Recipient address rejected: Access denied. AS(201806281) [SN1NAM02FT0043.eop-nam02.prod.protection.outlook.com]", "subcategory": "2003", "step_id": 489, "choice_number": 623}, "emitted_at": 1674507111307}
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-marketo/integration_tests/expected_records.jsonl:151]: 👍 fake emails (harry potter)

**Review by evantahler — APPROVED** [review]
👍 looking for PII

**Pull request merged**